R/initialisers.R

```diff
@@ -0,0 +1,67 @@
+#' init_analytics
+#' @description
+#' Creates the google-analytics.html script in order to allow the activation of
+#' analytics via GA4. For the full steps required to set up analytics, please
+#' refer to the documentation in the readme.
+#'
+#' @param ga_code The Google Analytics code
+#' @importFrom magrittr %>%
+#' @return TRUE if written, FALSE if not
+#' @export
+#'
+#' @examples init_analytics(ga_code = "0123456789")
+init_analytics <- function(ga_code) {
+  is_valid_ga4_code <- function(ga_code) {
+    stringr::str_length(ga_code) == 10 & typeof(ga_code) == "character"
+  }
+
+  if (is_valid_ga4_code(ga_code) == FALSE) {
+    stop(
+      'You have entered an invalid GA4 code in the ga_code argument.
+      Please enter a 10 digit code as a character string.
+      e.g. "0123QWERTY"'
+    )
+  }
+
+  webpage <- RCurl::getURL("https://raw.githubusercontent.com/dfe-analytical-services/dfeshiny/analytsics-init/inst/google-analytics.hml")
+  tryCatch(
+  html_script <- gsub(
+    "XXXXXXXXXX",
+    ga_code,
+    readLines(tc <- textConnection(webpage))
+  ),
+  error = function(e) {
+    return("Download failed")
+  },
+  message("Downloaded analytics template script")
+  )
+
+  close(tc)
+  if (file.exists("google-analytics.html")) {
+    message("Analytics file already exists.")
+    message("If you have any customisations in that file, make sure you've
+    backed those up before over-writing.")
+    user_input <- stringr::str_trim(
+    readline(
+      prompt = "Are you happy to overwrite the existing analytics script (y/N) "
+      )
+    )
+    if (user_input %in% c("y", "Y")) {
+      write_out <- TRUE
+    } else {
+      write_out <- FALSE
+    }
+  } else {
+    write_out <- TRUE
+  }
+  if (write_out) {
+    cat(html_script, file = "google-analytics.html")
```

Comment:
Is there a way to make this write out the html as separate lines? `writeLines()` or something similar.

If not then not a big deal, but currently it's putting it all as one line 
![image](https://github.com/user-attachments/assets/b8e5e945-6322-4250-aa2f-530b4438dd4b)

Reply:
Sorted